### PR TITLE
URLs Fixed by adding only $ and ^ sign

### LINKED ISSUE #1216: Add ^ & $ in regex of urls for proper matching
' `^` ' sign should be introduced in the regular expressions of the url's, mentioned in the url files so that our search is always matched up correctly.

For example:  
If we have not added the initial caret character (i.e., 'hello/$'), Django would match any URL that ends with hello/, such as /my/site/hello/.
Also it will be more syntactically correct.

### PR DESCRIPTION
**Pull request by HargovindArora**
Fix #1216 
Added `^` sign and only `$` sign as and where required in all Application's Urls.py files.

### CODE REVIEW

**Comment by HargovindArora** [comment]
@RishabhJain2018 Please see that why the test is failing, because this time only `$` sign is added at required places!

File: apps/challenges/urls.py [hunk 1]
```diff
@@ -4,38 +4,38 @@
...
-    url(r'challenge/challenge_host_team/(?P<challenge_host_team_pk>[0-9]+)/zip_upload/$',
+    url(r'^challenge/challenge_host_team/(?P<challenge_host_team_pk>[0-9]+)/zip_upload/$',
         views.create_challenge_using_zip_file,
         name='create_challenge_using_zip_file'),
-    url(r'(?P<challenge_pk>[0-9]+)/challenge_phase/(?P<challenge_phase_pk>[0-9]+)/submissions$',
+    url(r'^(?P<challenge_pk>[0-9]+)/challenge_phase/(?P<challenge_phase_pk>[0-9]+)/submissions$',
         views.get_all_submissions_of_challenge,
         name='get_all_submissions_of_challenge'),
-    url(r'(?P<challenge_pk>[0-9]+)/phase/(?P<challenge_phase_pk>[0-9]+)'
+    url(r'^(?P<challenge_pk>[0-9]+)/phase/(?P<challenge_phase_pk>[0-9]+)$'
```
Comment (RishabhJain2018) [inline apps/challenges/urls.py]: @HargovindArora Please check as we already have `$` at the end of this url on line 35.

File: apps/jobs/urls.py [hunk 2]
```diff
@@ -3,16 +3,16 @@
 from . import views
 
 urlpatterns = [
-    url(r'challenge/(?P<challenge_pk>[0-9]+)/'
+    url(r'^challenge/(?P<challenge_pk>[0-9]+)/$'
```
Comment (RishabhJain2018) [inline apps/jobs/urls.py]: Same as above.

File: apps/jobs/urls.py [hunk 3]
```diff
@@ -3,16 +3,16 @@
...
         r'challenge_phase/(?P<challenge_phase_pk>[0-9]+)/submission/(?P<submission_pk>[0-9]+)',
         views.change_submission_data_and_visibility, name='change_submission_data_and_visibility'),
-    url(r'challenge/(?P<challenge_id>[0-9]+)/'
+    url(r'^challenge/(?P<challenge_id>[0-9]+)/$'
```
Comment (RishabhJain2018) [inline apps/jobs/urls.py]: Same as above.

File: apps/jobs/urls.py [hunk 4]
```diff
@@ -3,16 +3,16 @@
...
         r'challenge_phase/(?P<challenge_phase_id>[0-9]+)/submission/',
         views.challenge_submission, name='challenge_submission'),
-    url(r'(?P<challenge_pk>[0-9]+)/'
+    url(r'^(?P<challenge_pk>[0-9]+)/$'
```
Comment (RishabhJain2018) [inline apps/jobs/urls.py]: Same as above.

Comment (HargovindArora) [reply apps/jobs/urls.py]: @RishabhJain2018 I didn't saw it, sorry my mistake.

**Comment by deshraj** [comment]
@HargovindArora seems like there is a merge conflict here. Can you please resolve it so that I can merge it?

**Comment by RishabhJain2018** [comment]
@HargovindArora Can you please resolve the merge conflicts.

**Comment by HargovindArora** [comment]
@deshraj Yeah. Doing it!

**Comment by HargovindArora** [comment]
@deshraj @RishabhJain2018 Does it looks good now?

**Comment by HargovindArora** [comment]
@deshraj @RishabhJain2018 Please review it!

**Pull request merged**